### PR TITLE
[WIP] DO NOT MERGE Trying if the CI is green with Kafka stuff in it

### PR DESCRIPTION
**Pull request by aliok**
/hold

Just trying if the upstream eventing-contrib e2e tests pass on OpenShift with Serverless Operator.

### CODE REVIEW

**Comment by aliok** [comment]
Result of 
```
pwd

  ls -laht

  touch test.yaml
  touch tmp.tracing.config.yaml
```

```
/go/src/knative.dev/eventing-contrib
total 284K
drwxr-xr-x. 21 root root 4.0K Oct  7 21:44 .
drwxr-xr-x.  5 root root   77 Oct  7 21:44 ..
drwxrwxr-x.  8 root root  210 Oct  5 19:52 .git
drwxrwxr-x. 17 root root 4.0K Oct  5 19:52 vendor
-rw-rw-r--.  1 root root  692 Oct  5 19:52 .gitattributes
drwxrwxr-x.  3 root root  101 Oct  5 19:52 .github
-rw-rw-r--.  1 root root   67 Oct  5 19:52 .gitignore
-rw-rw-r--.  1 root root   96 Oct  5 19:52 .ko.yaml
-rw-rw-r--.  1 root root  295 Oct  5 19:52 AUTHORS
-rw-rw-r--.  1 root root  168 Oct  5 19:52 CODE_OF_CONDUCT.md
-rw-rw-r--.  1 root root  210 Oct  5 19:52 CONTRIBUTING.md
-rw-rw-r--.  1 root root 4.2K Oct  5 19:52 DEVELOPMENT.md
-rw-rw-r--.  1 root root  12K Oct  5 19:52 LICENSE
-rw-rw-r--.  1 root root 3.0K Oct  5 19:52 Makefile
-rw-rw-r--.  1 root root  212 Oct  5 19:52 OWNERS
-rw-rw-r--.  1 root root  925 Oct  5 19:52 OWNERS_ALIASES
-rw-rw-r--.  1 root root 1.5K Oct  5 19:52 README.md
drwxrwxr-x.  6 root root   95 Oct  5 19:52 awssqs
drwxrwxr-x.  3 root root   46 Oct  5 19:52 camel
drwxrwxr-x.  5 root root   80 Oct  5 19:52 ceph
drwxrwxr-x.  7 root root  135 Oct  5 19:52 cmd
drwxrwxr-x.  3 root root   27 Oct  5 19:52 config
drwxrwxr-x.  3 root root   67 Oct  5 19:52 couchdb
drwxrwxr-x.  6 root root  116 Oct  5 19:52 github
drwxrwxr-x.  6 root root  137 Oct  5 19:52 gitlab
-rw-rw-r--.  1 root root 2.3K Oct  5 19:52 go.mod
-rw-rw-r--.  1 root root 205K Oct  5 19:52 go.sum
drwxrwxr-x.  2 root root  235 Oct  5 19:52 hack
drwxrwxr-x.  6 root root  167 Oct  5 19:52 kafka
drwxrwxr-x.  6 root root   92 Oct  5 19:52 natss
drwxrwxr-x.  4 root root  128 Oct  5 19:52 openshift
drwxrwxr-x.  6 root root  113 Oct  5 19:52 prometheus
drwxrwxr-x.  4 root root  157 Oct  5 19:52 samples
drwxrwxr-x.  8 root root 4.0K Oct  5 19:52 test
drwxrwxr-x.  3 root root   36 Oct  5 19:52 third_party
touch: cannot touch 'test.yaml': Permission denied
touch: cannot touch 'tmp.tracing.config.yaml': Permission denied
```

We need to change this 
```
sed "s/namespace: ${KNATIVE_DEFAULT_NAMESPACE}/namespace: ${TEST_EVENTING_NAMESPACE}/g" ${CONFIG_TRACING_CONFIG} > /tmp.tracing.config.yaml
  oc replace -f tmp.tracing.config.yaml
  rm tmp.tracing.config.yaml
```
in https://github.com/openshift/knative-eventing-contrib/blob/release-v0.17.1/openshift/e2e-common.sh#L115

to 
```
sed "s/namespace: ${KNATIVE_DEFAULT_NAMESPACE}/namespace: ${TEST_EVENTING_NAMESPACE}/g" ${CONFIG_TRACING_CONFIG} > /tmp/tmp.tracing.config.yaml
  oc replace -f /tmp/tmp.tracing.config.yaml
  rm /tmp/tmp.tracing.config.yaml
```

**Comment by aliok** [comment]
```
=== CONT  TestKafkaBinding/no_event-v1alpha1
    monitoring.go:78: Executing command: kubectl port-forward e2e-kafka-binding-event-logger 58522:8392 -n test-kafka-binding-no-event-v1alpha1-b4g2j
    monitoring.go:71: running e2e-kafka-binding-event-logger port-forward in background, pid = 16411
    event_info_store.go:114: Failed to start the EventInfoStore associated to pod 'e2e-kafka-binding-event-logger': timeout waiting for recordevents pod to come up: http get error: Get "http://localhost:58522/minmax": EOF
```

**Comment by aliok** [comment]
`event-logger` pod cannot start:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/520/pull-ci-openshift-knative-serverless-operator-master-4.5-upstream-e2e-aws-ocp-45/1314266071825387520/artifacts/upstream-e2e-aws-ocp-45/serverless-e2e/build-V2U3WEt6/pod-logs/test-kafka-binding-no-event-v1alpha1-b4g2j/e2e-kafka-binding-event-logger-recordevents.log
```
2020/10/08 19:21:33 maxprocs: Leaving GOMAXPROCS=2: CPU quota undefined
2020/10/08 19:21:33 Failed to process env varrequired key POD_NAME missing value
```

wtf?

**Comment by aliok** [comment]
@mgencur 
Here I am trying to run the eventing-contrib upstream (midstream) e2e tests on s-o.

Any help is appreciated!

https://github.com/openshift/knative-eventing-contrib/pull/578 is also failing because of the same problem. It is 0.17 eventing-contrib rebuild with very little changes.

There's an option to override the image used with `imagetemplate`, https://github.com/knative/pkg/blob/master/test/README.md#specifying-image-template, but not sure if the other tag in quay.io, v0.14.0, would work (gonna try).

**Comment by aliok** [comment]
/retest

Previous error (`2020/10/08 19:21:33 Failed to process env varrequired key POD_NAME missing value`) is gone.

Not sure what the new one is. Maybe a flake

**Comment by aliok** [comment]
another wtf moment:
```
{"level":"info","ts":"2020-10-10T13:46:07.337Z","logger":"mt-broker-controller.knative.dev-eventing-pkg-reconciler-mtbroker.Reconciler","caller":"mtbroker/broker.go:286","msg":"Created Channel: test-event-transformation-for-trigger-v1-beta1-broker-v1-b44nh2/kafkachannel-kne-trigger{NewChannel 22 0  0xc00068c660}","knative.dev/traceid":"aae0ed58-be40-4103-80dd-268dfb498ded","knative.dev/key":"test-event-transformation-for-trigger-v1-beta1-broker-v1-b44nh2/kafkachannel"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x166cef2]

goroutine 153 [running]:
knative.dev/eventing/pkg/reconciler/mtbroker.(*Reconciler).reconcileKind(0xc0002fe5b0, 0x1e49e00, 0xc0005bce40, 0xc00078edc0, 0xc00014b6f0, 0x1a7dfe9, 0x5)
	/go/src/knative.dev/eventing/pkg/reconciler/mtbroker/broker.go:159 +0xa22
knative.dev/eventing/pkg/reconciler/mtbroker.(*Reconciler).ReconcileKind(0xc0002fe5b0, 0x1e49e00, 0xc0005bce40, 0xc00078edc0, 0xc00078edc0, 0x0)
	/go/src/knative.dev/eventing/pkg/reconciler/mtbroker/broker.go:106 +0x6a
knative.dev/eventing/pkg/client/injection/reconciler/eventing/v1beta1/broker.(*reconcilerImpl).Reconcile(0xc0005e1d90, 0x1e49e00, 0xc0005bce10, 0xc0005b8320, 0x4c, 0xc00061af40, 0x1e49e00)
	/go/src/knative.dev/eventing/pkg/client/injection/reconciler/eventing/v1beta1/broker/reconciler.go:250 +0x1356
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc00035cff0, 0x0)
	/go/src/knative.dev/eventing/vendor/knative.dev/pkg/controller/controller.go:465 +0x785
knative.dev/pkg/controller.(*Impl).RunContext.func3(0xc0005be470, 0xc00035cff0)
	/go/src/knative.dev/eventing/vendor/knative.dev/pkg/controller/controller.go:403 +0x53
created by knative.dev/pkg/controller.(*Impl).RunContext
	/go/src/knative.dev/eventing/vendor/knative.dev/pkg/controller/controller.go:401 +0x1f5
```

happens in mt-broker-controller

File: hack/lib/serverless.bash [hunk 1]
```diff
@@ -151,6 +153,32 @@ EOF
...
+kind: KnativeKafka
+metadata:
+  name: knative-kafka
+  namespace: ${EVENTING_NAMESPACE}
+spec:
+  source:
+    enabled: true
+  channel:
+    enabled: true
+    bootstrapServers: my-cluster-kafka-bootstrap.kafka:9092
```
Comment (matzew) [inline hack/lib/serverless.bash:174]: we have Kafka, do we?

File: knative-operator/deploy/resources/knativekafka/kafkachannel-v0.17.1.yaml [hunk 2]
```diff
@@ -417,23 +417,6 @@ data:
   bootstrapServers: REPLACE_WITH_CLUSTER_URL
 ---
 apiVersion: v1
-kind: ConfigMap
```
Comment (matzew) [inline knative-operator/deploy/resources/knativekafka/kafkachannel-v0.17.1.yaml:420]: that sould have been removed on a different PR? :eyes:

Comment (aliok) [reply knative-operator/deploy/resources/knativekafka/kafkachannel-v0.17.1.yaml:420]: Yeah, https://github.com/openshift-knative/serverless-operator/pull/536

I am not doing things sequentially. I just rebased this PR on that.

File: hack/lib/serverless.bash (hunk 1 above)
Comment (aliok) [reply hack/lib/serverless.bash:174]: Yeah, installed  here: https://github.com/openshift-knative/serverless-operator/pull/520/files#diff-196879e6787eb2045a7c3ca7eac896b28577a5b8cea9eeae66b835a145616220R36

**Comment by aliok** [comment]
Only `TestKafkaBinding` tests failed this time, rest passed.

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/520/pull-ci-openshift-knative-serverless-operator-master-4.5-upstream-e2e-aws-ocp-45/1316087399083675648/artifacts/upstream-e2e-aws-ocp-45/serverless-e2e/build-nBj9SEyQ/pod-logs/test-kafka-binding-no-event-v1alpha1-gb9x8/2a9b554e-8b0d-4519-95f1-44c9cdc878a6-0-producer-rrz9r-2a9b554e-8b0d-4519-95f1-44c9cdc878a6-0-producer-container.log

```
2020/10/13 20:04:53 GOT: {2a9b554e-8b0d-4519-95f1-44c9cdc878a6 0 map[content-type:application/json] {"value":5}}
2020/10/13 20:05:27 kafka: client has run out of available brokers to talk to (Is your cluster reachable?)
```

I know Kafka cluster is available... Perhaps the check to see if Kafka is ready is not good. It just checks the pods. Better would be checking the `kafka` CR's status. Might do this later.

/retest

**Pull request closed without merging**